features/dd-sdk-android-trace/src/test/kotlin/com/datadog/android/trace/impl/DatadogTracerAdapterTest.kt

```diff
@@ -0,0 +1,141 @@
+import fr.xgouchet.elmyr.annotation.StringForgery
+import fr.xgouchet.elmyr.junit5.ForgeConfiguration
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+import org.mockito.Mock
+import org.mockito.junit.jupiter.MockitoExtension
+import org.mockito.junit.jupiter.MockitoSettings
+import org.mockito.kotlin.any
+import org.mockito.kotlin.isA
+import org.mockito.kotlin.mock
+import org.mockito.kotlin.verify
+import org.mockito.kotlin.whenever
+import org.mockito.quality.Strictness
+
+@Extensions(
+    ExtendWith(MockitoExtension::class),
+    ExtendWith(ForgeExtension::class)
+)
+@MockitoSettings(strictness = Strictness.LENIENT)
+@ForgeConfiguration(Configurator::class)
+internal class DatadogTracerAdapterTest {
+
+    private lateinit var testedTracer: DatadogTracerAdapter
+
+    @Mock
+    lateinit var mockSdk: FeatureSdkCore
+
+    @Mock
+    lateinit var mockTracer: AgentTracer.TracerAPI
+
+    @Mock
+    lateinit var mockSpan: AgentSpan
+
+    @Mock
+    lateinit var mockDatadogSpan: DatadogSpanAdapter
+
+    @Mock
+    lateinit var mockScope: AgentScope
+
+    @Mock
+    lateinit var mockSpanBuilder: AgentTracer.SpanBuilder
+
+    @StringForgery
+    lateinit var fakeString: String
+
+    @BoolForgery
+    private var fakeBool: Boolean = false
+
+    @BeforeEach
+    @Suppress("DEPRECATION")
+    fun `se tup`() {
+        testedTracer = DatadogTracerAdapter(mockSdk, mockTracer, true)
+        whenever(mockDatadogSpan.delegate).thenReturn(mockSpan)
+        whenever(mockTracer.propagate()).thenReturn(mock())
+        whenever(mockTracer.buildSpan(any())).thenReturn(mockSpanBuilder)
+        whenever(mockTracer.buildSpan(any(), any())).thenReturn(mockSpanBuilder)
+    }
+
+    @Test
+    @Suppress("DEPRECATION")
+    fun `M return span W buildSpan(String)`() {
+        val builder = testedTracer.buildSpan(fakeString)
+
+        assertThat(builder).isInstanceOf(DatadogSpanBuilder::class.java)
+        verify(mockTracer).buildSpan(fakeString)
+    }
+
+    @Test
+    fun `M return span W buildSpan(String, String) `() {
+        val builder = testedTracer.buildSpan(fakeString, fakeString)
+
+        assertThat(builder).isInstanceOf(DatadogSpanBuilder::class.java)
+        verify(mockTracer).buildSpan(fakeString, fakeString)
+    }
+
+    @Test
+    fun `M delegate SpanBuilder#addScopeListener W addScopeListener`() {
+        val mockListener = mock<DataScopeListener>()
+
+        testedTracer.addScopeListener(mockListener)
+
+        verify(mockTracer).addScopeListener(isA<DatadogScopeListenerAdapter>())
+    }
+
+    @Test
+    fun `M return wrapped span W activeSpan`() {
+        whenever(mockTracer.activeSpan()).thenReturn(mockSpan)
+
+        val span = testedTracer.activeSpan() as DatadogSpanAdapter
```

Comment:
seems like the case here and below is redundant, since `assertThat` call below is just for the generic object, so `isEqualTo` should work without any cast as well.

If we want to check the type explicitly, we need to use `isInstanceOf`